operations/helm/charts/grafana-agent/values.yaml

```diff
@@ -152,6 +152,9 @@ controller:
   # -- Number of pods to deploy. Ignored when controller.type is 'daemonset'.
   replicas: 1
 
+  # -- Annotations to add to controller.
+  annotations: {}
```

Comment:
For future flexibility, should this be `extraAnnotations`? (For example, if we eventually add a default set of annotations, this should likely append instead of overwrite)

Reply:
every time I've seen this parameter exposed in a helm chart it was just named "annotations".

Reply:
For example: https://github.com/grafana/helm-charts/blob/main/charts/grafana/values.yaml#L190-L191

Reply:
Looking through this chart, we do currently use the `extra*` terminology a few times in our chart:

* `agent.extraEnv`
* `agent.extraArgs`
* `agent.extraPorts`
* `agent.mounts.extra`
* `controller.volumes.extra` 
* `ingress.extraPaths`

There are a few cases where, like you mentioned, it's not used:

* `controller.podAnnotations`
* `controller.podLabels` 

I'm wondering if it makes sense to be consistent here, especially since the current chart has an implication of being additive whenever "extra" is mentioned.

Reply:
I don't mind to change `annotations` to `extraAnnotations`. @rfratto could you please discuss that with other maintainers of the Helm Chart so that I can make a change if needed? Or maybe you are the only one to decide, then please tell me what is your preference.

Reply:
In this case, we are not adding any annotations of our own, so I don't know that `extra` is needed. But one common pitfal is that people put annotations on the controller, expecting them to show up on the pod as well. Whatever we end up with should help people with that distinction. I am ok with `annotations`, with a reasonable comment.

Reply:
> In this case, we are not adding any annotations of our own, so I don't know that extra is needed

It's not in this particular case, but my original comment is concerned about future extension, to allow us flexibility in the future to start adding pod annotations if we need to.

Reply:
I like the `extra*` convention for things we extend/append to, and make sure we don't use it for things we replace instead.

Reply:
@rfratto @captncraig @tpaschalis PR amended